operator/v1/types_machineconfiguration.go

```diff
@@ -56,3 +60,21 @@ type MachineConfigurationList struct {
 	// Items contains the items
 	Items []MachineConfiguration `json:"items"`
 }
+
+// ManagedBootImageConfig contains configuration data for the bootimages management feature
+// +union
+type ManagedBootImageConfig struct {
```

Comment:
Needs CEL expression to make sure that `MatchSelector` is only set when type is `MatchSelector`. See the `example/v1` `CELUnion` type for an example